posts/2023-11-05-telegram-bot-for-file-sharing-using-quarkus.adoc

```diff
@@ -0,0 +1,546 @@
+        }
+    }
+----
+Yes, hardcode it. Yes, "if" is everywhere (2).
+
+In the case of media attachment, it will expect a file, photo, or video.
+[source,java]
+----
+    public void handleMedia(Message message) throws TelegramApiException, IOException {
+        var chatId = message.getChatId();
+        LOGGER.infof("Handled media in chatId: %s", chatId);
+
+        String fileName;
+        String fileId;
+        String filePath;
+        if (message.getDocument() != null) {
+            fileName = message.getDocument().getFileName();
+            fileId = message.getDocument().getFileId();
+            filePath = telegramLongPollingBot.execute(new GetFile(fileId)).getFilePath();
+        } else if (message.getPhoto() != null) {
+            fileName = "photo_" + LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) + ".jpeg";
+            fileId = Collections.max(message.getPhoto(), Comparator.comparing(PhotoSize::getFileSize)).getFileId();
+            filePath = telegramLongPollingBot.execute(new GetFile(fileId)).getFilePath();
+        } else if (message.getVideo() != null) {
+            fileName = message.getVideo().getFileName();
+            fileId = message.getVideo().getFileId();
+            filePath = telegramLongPollingBot.execute(new GetFile(fileId)).getFilePath();
+        } else {
+            LOGGER.errorf("Error while handling media in chatId: :s", chatId);
+            throw new RuntimeException("Error while handling media in chatId: " + chatId);
+        }
+
+        var bytes = telegramLongPollingBot.downloadFileAsStream(filePath).readAllBytes();
+        minioService.uploadFile(String.valueOf(chatId), fileName, bytes);
+    }
+----
+Yes, hardcode it (2). Yes, "if" is everywhere (3).
+
+For interaction with MinIO we have to create a client bean like we already did it.
+[source,java]
+----
+    @Produces
+    public MinioClient minioClient() {
+        try {
+            return MinioClient.builder()
+                    .endpoint(
+                            minIOConfig.host(),
+                            minIOConfig.port(),
+                            minIOConfig.useSsl())
+                    .credentials(
+                            minIOConfig.accessKey(),
+                            minIOConfig.secretKey())
+                    .build();
+        } catch (Exception e) {
+            LOGGER.errorf(e, "Error while initializing minioClient");
+            throw new RuntimeException(e);
+        }
+    }
+----
+
+MinIO is a high-performance object storage server. MinIO's core concept is a "bucket" identified by a unique bucket ID. All files associated with a bucket are stored kinda together as a group. In our case, we'll use the chat ID as the bucket ID, which is unique for each Telegram user. This way, we can easily retrieve files by specifying the bucket ID and their respective names.
+
+[source,java]
+----
+    private byte[] getFileAsBytes(String bucketName, String objectName) {
```

Comment:
if the `bucketName` is what you  just called of "unique bucket ID" then it would be more clear if in the code you used the variable name `bucketId`

Reply:
In the java MinIO client library, it's a bucket "name". So, changed it to "name". Thanks!